src/main/java/uk/gov/pay/api/filter/AuthorizationValidationFilter.java

```diff
@@ -38,8 +40,11 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
         final String authorization = ((HttpServletRequest) request).getHeader("Authorization");
 
         if (isValidAuthorizationHeader(authorization)) {
-            chain.doFilter(request, response);
-
+            try {
+                chain.doFilter(request, response);
+            } finally {
+                MDC.remove(GATEWAY_ACCOUNT_ID);
```

Comment:
This seems rather incongruent to me - should we define a dedicated filter for managing request-lifecycle MDC keys?

Reply:
Yeah I think thats a reasonable suggestion. We could create an MDC Filter and give it priority so it is at the edge of our app and clear the MDC values there. 
```java
try {
        chain.doFilter(request, response);
} finally {
         MDC.clear();
}
```

I'll make the change 👍